pkg/controller/servicemesh/webhooks/mutation/controlplane.go

```diff
@@ -299,3 +320,29 @@ func (m *smcpv2mutator) SetOpenShiftRouteEnabled(value bool) {
 
 	m.patches = append(m.patches, jsonpatch.NewPatch("add", "/spec/gateways", *gateways))
 }
+
+func (m *smcpv1mutator) IsTracingTypeSpecified() *bool {
+	return nil
+}
+
+func (m *smcpv1mutator) SetTracingType(value string) {}
+
+func (m *smcpv2mutator) IsTracingTypeSpecified() *bool {
+	tracingType := m.smcp.Spec.Tracing.Type
+
+	if tracingType == nil {
+		return nil
+	}
+
+	route := gateways.OpenShiftRoute
+
+	if route == nil {
+		return nil
+	}
+
+	return route.Enabled
+}
+
+func (m *smcpv2mutator) SetTracingType(value string) {
+	m.patches = append(m.patches, jsonpatch.NewPatch("add", "/spec/gateways/openshiftRoute/enabled", value))
```

Comment:
`"/spec/gateways/openshiftRoute/enabled"` should be changed accordingly. Setting `/spec/tracing/type` will fail if `/spec/tracing` is `nil`.